analyzers/src/SonarAnalyzer.CFG/ShimLayer/AnalysisContext/SymbolStartAnalysisContext.cs

```diff
@@ -0,0 +1,169 @@
+        registerOperationBlockAction(RoslynSymbolStartAnalysisContext, action);
+
+    public void RegisterOperationBlockStartAction(Action<OperationBlockStartAnalysisContext> action) =>
+        registerOperationBlockStartAction(RoslynSymbolStartAnalysisContext, action);
+
+    public void RegisterSymbolEndAction(Action<SymbolAnalysisContext> action) =>
+        registerSymbolEndAction(RoslynSymbolStartAnalysisContext, action);
+
+    public void RegisterSyntaxNodeAction<TLanguageKindEnum>(Action<SyntaxNodeAnalysisContext> action, params TLanguageKindEnum[] syntaxKinds) where TLanguageKindEnum : struct
+    {
+        var languageKindType = typeof(TLanguageKindEnum);
+        if (languageKindType == typeof(CS.SyntaxKind))
+        {
+            registerSyntaxNodeActionCS(RoslynSymbolStartAnalysisContext, action, syntaxKinds.Cast<CS.SyntaxKind>().ToImmutableArray());
+        }
+        else if (languageKindType.FullName == "Microsoft.CodeAnalysis.VisualBasic.SyntaxKind")
+        {
+            throw new NotImplementedException("Add a reference to the Microsoft.CodeAnalysis.VisualBasic.Workspaces package.");
+        }
+        else
+        {
+            throw new ArgumentException("Invalid type parameter.", nameof(TLanguageKindEnum));
+        }
+    }
+
+    // symbolStartAnalysisContextParameter => ((symbolStartAnalysisContextType)symbolStartAnalysisContextParameter)."propertyName"
+    private static Func<object, TProperty> CreatePropertyAccessor<TProperty>(Type symbolStartAnalysisContextType, string propertyName)
+    {
+        var symbolStartAnalysisContextParameter = Parameter(typeof(object));
+        return Lambda<Func<object, TProperty>>(
+            Property(
+                Convert(symbolStartAnalysisContextParameter, symbolStartAnalysisContextType), propertyName),
+            symbolStartAnalysisContextParameter).Compile();
+    }
+
+    // (object symbolStartAnalysisContextParameter, Action<TContext> registerActionParameter)
+    //     => ((symbolStartAnalysisContextType)symbolStartAnalysisContextParameter)."registrationMethodName"<typeArguments>(registerActionParameter)
+    private static Action<object, Action<TContext>> CreateRegistrationMethod<TContext>(Type symbolStartAnalysisContextType, string registrationMethodName, params Type[] typeArguments)
+    {
+        var receiver = Parameter(typeof(object));
+        var registerActionParameter = Parameter(typeof(Action<TContext>));
+        return Lambda<Action<object, Action<TContext>>>(
+            Call(Convert(receiver, symbolStartAnalysisContextType), registrationMethodName, typeArguments, registerActionParameter), receiver, registerActionParameter).Compile();
+    }
+
+    // (object symbolStartAnalysisContextParameter, Action<TContext> registerActionParameter, TParameter additionalParameter)
+    //     => ((symbolStartAnalysisContextType)symbolStartAnalysisContextParameter)."registrationMethodName"<typeArguments>(registerActionParameter, additionalParameter)
```

Comment:
```suggestion
    // (object obj, Action<TContext> registerAction, TParameter additional) =>
    //     ((symbolStartAnalysisContextType)obj)."registrationMethodName"<typeArguments>(registerAction, additional)
```

(`obj` instead of `symbolStartAnalysisContextParameter`, `Parameter` suffix removed, `=>` on previous line, as per .NET Coding Style guidelines).

Reply:
naming of parameter is good, but `=>` should go to the previous line.